arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -413,110 +414,93 @@ static inline bool isStopping() {
   return server.isStopping();
 }
 
-arangodb::Result SynchronizeShard::getReadLock(std::string const& endpoint,
-                                               std::string const& database,
-                                               std::string const& collection,
-                                               std::string const& clientId, uint64_t rlid,
-                                               bool soft, double timeout) {
+arangodb::Result SynchronizeShard::getReadLock(
+  std::string const& endpoint, std::string const& database,
+  std::string const& collection, std::string const& clientId,
+  uint64_t rlid, bool soft, double timeout) {
+
+  // This function can be implemented in a more robust manner for server
+  // versions > 3.4. Starting with 3.4 the POST requests to the read lock API
+  // terminates the server side thread as soon as the lock request comes in.
+  // The POST request thus is answered immediately back to the caller.
+  // The servers (<=3.3) with lower versions hold the POST request for as long
+  // as the corresponding DELETE_REQ has not been successfully submitted.
+  
+  using namespace std::chrono;
+  auto const start = steady_clock::now();
   auto cc = arangodb::ClusterComm::instance();
-  if (cc == nullptr) {  // nullptr only happens during controlled shutdown
-    return arangodb::Result(TRI_ERROR_SHUTTING_DOWN,
-                            "startReadLockOnLeader: Shutting down");
+
+  // nullptr only happens during controlled shutdown
+  if (cc == nullptr) {  
+    return arangodb::Result(
+      TRI_ERROR_SHUTTING_DOWN, "startReadLockOnLeader: Shutting down");
   }
 
   VPackBuilder body;
-  {
-    VPackObjectBuilder o(&body);
+  { VPackObjectBuilder o(&body);
     body.add(ID, VPackValue(std::to_string(rlid)));
     body.add(COLLECTION, VPackValue(collection));
     body.add(TTL, VPackValue(timeout));
-    body.add(StaticStrings::ReplicationSoftLockOnly, VPackValue(soft));
-  }
-
-  auto url = DB + database + REPL_HOLD_READ_LOCK;
+    body.add(REBOOT_ID, VPackValue(ServerState::instance()->getRebootId()));
```

Comment:
I overlooked this before - originally, we planned to send the reboot id in a header with every request iirc. If we still plan to do that, we should do it from the beginning. If that doesn't work (what about Vst, anyway?) or we decided against it for some reason, this is fine. @mchacki @neunhoef

Reply:
good point, also overlooked this.
Personally I Don not have much preference, but it should be consistent, either all body or all headers.
AFAIK VST supports headers.

Using a header for it, we could simply attach it to every request in the communication.

Reply:
should this go into this PR? is the only thing i am wondering